plugins/module_utils/network/sonic/config/dcbx/dcbx.py

```diff
@@ -0,0 +1,638 @@
+                    intf.pop('enabled')
+                    pop_item = True
+                if 'name' in intf and len(intf) == 1 and pop_item:
+                    idx = intfs.index(intf)
+                    pop_list.insert(0, idx)
+
+            for idx in pop_list:
+                intfs.pop(idx)
+            if not intfs:
+                data.pop('interfaces')
+
+    def _state_deleted(self, want, have, diff):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        requests = []
+
+        is_delete_all = False
+        if not want:
+            commands = deepcopy(have)
+            is_delete_all = True
+        else:
+            commands = get_diff(want, diff, TEST_KEYS)
+        self.remove_default_entries(commands)
+
+        intf_commands = commands.get('interfaces')
+        if intf_commands:
+            commands['interfaces'] = remove_empties_from_list(intf_commands)
+
+        requests = self.get_delete_dcbx_requests(commands, have, is_delete_all)
+        if len(requests) == 0:
+            commands = []
+
+        if commands:
+            commands = update_states(commands, "deleted")
+        return commands, requests
+
+    def _state_replaced(self, diff, want, have):
+        """The command generator when state is replaced
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+        to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        commands, requests = self.get_commands_requests_for_replaced_overridden(diff, want, have, 'replaced')
+
+        return commands, requests
+
+    def _state_overridden(self, diff, want, have):
+        """The command generator when state is overridden
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+        to the desired configuration
+        """
+        commands = []
+        requests = []
+        commands, requests = self.get_commands_requests_for_replaced_overridden(diff, want, have, 'overridden')
+
+        return commands, requests
+
+    def get_commands_requests_for_replaced_overridden(self, diff, want, have, state):
+        """Returns the commands and requests necessary to remove applicable
+        current configurations when state is replaced or overridde and global_conf == defaultn
+        """
+        default_global_dcbx = {'enabled': False}
+        default_interface_dcbx = {
+            'pfc_tlv_enabled': True,
+            'ets_configuration_tlv_enabled': True,
+            'ets_recommendation_tlv_enabled': True,
+            'enabled': True
+        }
+        commands = []
+        requests = []
+        del_commands = []
+        add_commands = []
+
+        if not have:
+            commands = diff
```

Comment:
A change to a non-default value for the global configuration option needs to be handled for this block.

Reply:
In real case, "have" would not be empty. If it is "empty" for some reason, function get_modify_specific_dcbx_interfaces_param_requests already handles both global dcbx admin and interface( changed the function name to get_modify_specific_dcbx_global_and_interfaces_param_requests).